### PR TITLE
mvc-first-commit

### PR DESCRIPTION
**Pull request by Arserar**
I have added mvc structure and connected to sequelize.

### CODE REVIEW

File: config/db.config.js [hunk 1]
```diff
@@ -0,0 +1,13 @@
+module.exports = {
+    host: "localhost",
+    username: "postgres",
```
Comment (Karlen92) [inline config/db.config.js:3]: add DB connections to `.env `file

File: index.js [hunk 2]
```diff
@@ -1,16 +1,21 @@
...
 const PORT = process.env.PORT || 3000;
 
 app.use(express.json());
 app.use(express.urlencoded({ extended: true }));
 app.use('/api/user', userRoutes);
+app.use('/api', authRoutes);
 app.use(errorHandler);
 app.use(successHandler);
 
+sequelize.sync();
```
Comment (Karlen92) [inline index.js:17]: move this inside 
`app.listen(PORT, () => { ....`

The idea is that we need to do this when the project is started successfully.

File: models/index.js [hunk 3]
```diff
@@ -0,0 +1,11 @@
+const dbConfig = require('../config/db.config.js');
+
+const Sequelize = require('sequelize');
```
Comment (Karlen92) [inline models/index.js:3]: Did you use `sequlize init ` ? this file does not look like auto-generated `sequlize` file.

File: services/userService.js [hunk 4]
```diff
@@ -0,0 +1,72 @@
...
+};
+
+exports.getUserById = async (id) => {
+    return await User.findByPk(id);
+};
+
+exports.createUser = async (user) => {
+    try {
+        const hash = await bcrypt.hash(user.password, 10);
+        user.password = hash;
```
Comment (Karlen92) [inline services/userService.js:27]: never change parameters of the function, this is wrong

File: services/userService.js [hunk 5]
```diff
@@ -0,0 +1,72 @@
...
+    const user = await getUserByEmail(email);
+
+    if (!user) {
+        throw new HandleError('Please enter existing email!');
+    }
+
+    const match = await bcrypt.compare(password, user.password);
+
+    if (!match) {
+        throw new HandleError('Invalid credentials!');
```
Comment (Karlen92) [inline services/userService.js:65]: `HandleError` the name of the error is not saying anything about error, better use `BaseError` if nothing more specific

**Review by Karlen92 — CHANGES_REQUESTED** [review]

File: services/userService.js (hunk 4 above)
Comment (Arserar) [reply services/userService.js:27]: Can I do this - const newUser = {...user}; and then use newUser?

**Pull request merged**